source/robot_model/include/robot_model/Model.hpp

```diff
@@ -185,11 +225,21 @@ class Model {
                                         const state_representation::JointPositions& joint_positions,
                                         const std::vector<std::string>& frames);
 
+
+  /**
+   * @brief Generates a list of collision pairs to exclude based on the kinematic tree of the model
+   * @return the list of collision pairs to exclude 
+  */
+  std::vector<pinocchio::CollisionPair> generate_joint_exclusion_list();
+
 public:
   /**
    * @brief Constructor with robot name and path to URDF file
    */
-  explicit Model(const std::string& robot_name, const std::string& urdf_path);
+  explicit Model(const std::string& robot_name, 
+                 const std::string& urdf_path, 
+                 const bool load_collision_geometries = false,
+                 const std::function<std::string(const std::string&)>& package_paths_resolver_callback_ = nullptr);
```

Comment:
While this is not a breaking API change when it comes to the constructor (because of default arguments, the old form of the constructor can still be used), but I realized it _will_ constitute a breaking ABI change for the constructor symbols in the shared library binary. This means that we could only release this change as part of a version 8.0.0.


The alternative would be to keep the initial explicit constructor unchanged, and make a second constructor that resolves differently (i.e. explicit arguments with no default values). This would make it non-breaking for the ABI but otherwise very similar for the user.

```c++
explicit Model(const std::string& robot_name, const std::string& urdf_path);

explicit Model(const std::string& robot_name, 
                  const std::string& urdf_path,
                  const std::optional<std::function<std::string(const std::string&)>>& meshloader_callback);

// load a model with no meshes
Model(name, path);

// load a model with meshes with no path replacement necessary
Model(name, path, {});

// load a model with meshes with path replacement
Model(name, path, [](...){...});
```

Also a few nitpicks for the function callback: you used `package_paths_resolver_callback_` here but `meshloader_callback` everywhere else, so I would rename this. I would personally prefer `std::optional` wrapper around this, which makes it more clear from an API side. The default null function is essentially the same behavior but less explicit for the user.

Finally please also update the docstring for the constructor(s) (it had missing `@param` list in the first place)